### PR TITLE
Fixes some bugs with the detective's revolver.

### PR DESCRIPTION
**Pull request by uomo91**
## About The Pull Request

The detective revolver's ammo mod process used screwdriver_act which messed things up with the parent proc for firing pin removal, you had to remove the firing pin inside to do it, and it also spat out a message about removing the firing pin even if one was not present. This switches it to wrench_act to modify the ammo type, and further only lets you modify the gun's ammo type if you are holding it.

This also switches the gun from using after_attack to using process_fire when it misfires (from wrenching or from firing 357 ammo), fixing some ugly and and wrongly ordered chat outputs and a bug where when it misfired from firing 357 ammo it didn't expend the chambered round.

This also generally makes the det revolver's code much nicer and cleaner.

I was also able to get the gun to damage whatever hand it is being held in when 357 ammo causes a misfire, and to make it so you **only** drop the gun on a misfire that discharges a **live round**.

## Why It's Good For The Game

1623394153-Calvin_Hobbes-TracerBullet2

## Changelog
:cl:
fix: To change the ammo type the detective's revolver can fire you must now use a wrench instead of a screwdriver, removing the need to take out or destroy its firing pin, and you must be holding it to do so.
fix: When the detective's revolver misfires from using 357 ammo, the chambered round will actually be spent now.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 1]
```diff
@@ -92,48 +92,43 @@
 						"Black Panther" = "detective_panther"
 						)
 
-/obj/item/gun/ballistic/revolver/detective/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
-	if(magazine && magazine.caliber != initial(magazine.caliber))
+/obj/item/gun/ballistic/revolver/detective/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0, skip_missfire_check = FALSE)
```
Comment (Rohesie) [inline code/modules/projectiles/guns/ballistic/revolver.dm:95]: The arguments of redefined procs should be the same in parent and child.
Turn this into a detective revolver variable, that will be more maintainable.

Comment (uomo91) [reply code/modules/projectiles/guns/ballistic/revolver.dm:96]: This is done now I believe.

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 2]
```diff
@@ -91,49 +91,47 @@
 						"The Peacemaker" = "detective_peacemaker",
 						"Black Panther" = "detective_panther"
 						)
+	var/skip_357_missfire_check = FALSE
 
-/obj/item/gun/ballistic/revolver/detective/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
-	if(magazine && magazine.caliber != initial(magazine.caliber))
+/obj/item/gun/ballistic/revolver/detective/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0, skip_missfire_check = FALSE)
```
Comment (Rohesie) [inline code/modules/projectiles/guns/ballistic/revolver.dm:96]:
```suggestion
/obj/item/gun/ballistic/revolver/detective/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
```

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 3]
```diff
@@ -91,49 +91,47 @@
 						"The Peacemaker" = "detective_peacemaker",
 						"Black Panther" = "detective_panther"
 						)
+	var/skip_357_missfire_check = FALSE
```
Comment (Rohesie) [inline code/modules/projectiles/guns/ballistic/revolver.dm:94]: Document it?

File: code/modules/projectiles/guns/ballistic/revolver.dm (hunk 2 above)
Comment (uomo91) [reply code/modules/projectiles/guns/ballistic/revolver.dm:96]: Oops.

File: code/modules/projectiles/guns/ballistic/revolver.dm (hunk 3 above)
Comment (uomo91) [reply code/modules/projectiles/guns/ballistic/revolver.dm]: Alright.

**Pull request merged**